src/meta/server_state.cpp

```diff
@@ -1259,6 +1259,111 @@ void server_state::drop_app(dsn::message_ex *msg)
     }
 }
 
+void server_state::do_app_rename(configuration_rename_app_rpc rpc)
```

Comment:
How about rename to `rename_app_unlock`? the `_unlock` postfix is often used for the functions which must hold a lock by caller.